Tribler/Core/APIImplementation/LaunchManyCore.py

```diff
@@ -761,11 +761,9 @@ def network_shutdown(self):
 
     def save_download_pstate(self, infohash, pstate):
         """ Called by network thread """
-        basename = binascii.hexlify(infohash) + '.state'
-        filename = os.path.join(self.session.get_downloads_pstate_dir(), basename)
 
-        self._logger.debug("tlm: network checkpointing: to file %s", filename)
-        pstate.write_file(filename)
+        self.downloads[infohash].pstate_for_restart = pstate
```

Comment:
Because `pstate` is provided by different method and I'm not sure those will return the same value, so I keep the parameter `pstate` there and use it as a basis to save the state